src/backend/catalog/toasting.c

```diff
@@ -201,6 +201,9 @@ create_toast_table(Relation rel, Oid toastOid, Oid toastIndexOid,
 	if (sql_dialect == SQL_DIALECT_TSQL && RelationIsBBFTableVariable(rel))
 		pg_toast_prefix = "@pg_toast";
 
+	if (sql_dialect == SQL_DIALECT_TSQL && RelationIsBBFTempTable(rel))
+		pg_toast_prefix = "#pg_toast";
```

Comment:
I think it's not as simple as Table Variables. All Table Variables are in ENR, but not all temp tables are in ENR. Furthermore, temp table and its corresponding toast table may not have the same columns so you may end up with temp table not in ENR but toast table in ENR or vise versa. 

Consider the following tests which now fails with this code but clean before.
```
CREATE TYPE typa FROM int
GO

CREATE TYPE typb FROM nvarchar(100)
GO

-- Temp table wont be in ENR because of the `CheckTempTableHasDependencies` in `heap_create_with_catalog` but the toast table is in ENR. So you may need to add some changes to `heap_create_with_catalog`.
CREATE TABLE #temp_3(a typa, b typb)
GO

DROP TABLE #temp_3
GO

-- This fails now but it used to be clean.
CREATE TABLE #temp_3(a typa, b typb)
GO

DROP TYPE typa
DROP TYPE typb
GO
```